Sources/NetworkProtection/NetworkProtectionDeviceManager.swift

```diff
@@ -203,29 +219,18 @@ public actor NetworkProtectionDeviceManager: NetworkProtectionDeviceManagement {
                 errorEvents?.fire(NetworkProtectionError.serverListInconsistency)
 
                 let cachedServer = try cachedServer(registeredWith: keyPair)
-                return (cachedServer, keyPair)
+                return (cachedServer, nil)
             }
 
             selectedServer = registeredServer
-
-            // We should not need this IF condition here, because we know registered servers will give us an expiration date,
-            // but since the structure we're currently using makes the expiration date optional we need to have it.
-            // We should consider changing our server structure to not allow a missing expiration date here.
-            if let serverExpirationDate = selectedServer.expirationDate,
-               keyPair.expirationDate > serverExpirationDate {
-
-                keyPair = keyStore.updateCurrentKeyPair(newExpirationDate: serverExpirationDate)
-            }
-
-            return (selectedServer, keyPair)
+            return (selectedServer, selectedServer.expirationDate)
```

Comment:
The key pair is handled outside of this function now and passed into this func as an argument, so instead of being able to return a new one we just return a new expiration date, if one exists.